lib/vagrant-ovirt3/action/create_vm.rb

```diff
@@ -34,19 +34,30 @@ def call(env)
           env[:ovirt_cluster] = cluster
 
           # Get template
-          template = OVirtProvider::Util::Collection.find_matching(
-            env[:ovirt_compute].templates.all, config.template)
+          template = env[:ovirt_compute].templates.all.find_all { |t|
+            t.id == config.template or t.name == config.template
+          }
+          .sort_by { |t| t.raw.version.version_number.to_i }.reverse
+          .find { |t|
+            v = t.raw.version
+            cv = config.template_version
+            cv.nil? or (cv.to_i == v.version_number.to_i or cv == v.version_name)
+          }
           if template == nil
             raise Error::NoTemplateError,
               :template_name => config.template
           end
 
           # Output the settings we're going to use to the user
+          ver = template.raw.version
           env[:ui].info(I18n.t("vagrant_ovirt3.creating_vm"))
           env[:ui].info(" -- Name:          #{name}")
           env[:ui].info(" -- Cpus:          #{cpus}")
           env[:ui].info(" -- Memory:        #{memory_size/1024}M")
           env[:ui].info(" -- Template:      #{template.name}")
+          if ver.version_name or ver.version_number
+            env[:ui].info(" -- Version:       #{ver.version_name || ver.version_number}")
```

Comment:
What's the benefit to using version_number? It looks to be calculated by the engine. It looks as though it's just a fallback, but I think I'd rather just use it with name instead of one or the other such as *name (version)*  if name is given, or just version if not. I say this because [the docs show name is optional](http://www.ovirt.org/Features/Template_Versions).

perhaps...

```
if !template.raw.version.nil? and !template.raw.version.empty?
  version_string = "#{template.raw.version.version_name} (#{template.raw.version.version_number})"
else
  version_string = "#{template.raw.version.version_number}"
end
env[:ui].info(" -- Version:       #{version_string}")
```